validator/submission_tester/api.py

```diff
@@ -1,95 +1,46 @@
 import asyncio
 import logging
 import sys
-from asyncio import Future, AbstractEventLoop
-from collections.abc import Iterable
 from contextlib import asynccontextmanager
-from typing import TextIO
+from io import StringIO
+from queue import Queue
 
 from base_validator import API_VERSION
 from base_validator.metrics import BenchmarkState, BenchmarkResults
 from fastapi import FastAPI, WebSocket, Request
-
 from neuron import CURRENT_CONTEST, CheckpointSubmission, Key
-from .benchmarker import Benchmarker
-
-
-async def send_data(websocket: WebSocket, data: list[str]):
-    for line in data:
-        await websocket.send_text(line)
-
 
-class WebSocketLogStream:
-    _websocket: WebSocket
-    _log_type: str
-    _delegate: TextIO
+from .benchmarker import Benchmarker
 
-    _data: list[str]
-    _futures: list[Future[None]]
+logs = Queue()
 
-    _loop: AbstractEventLoop
+class LogsIO(StringIO):
+    old_stdout: StringIO
 
-    def __init__(self, websocket: WebSocket, log_type: str, delegate: TextIO, loop: AbstractEventLoop):
+    def __init__(self, old_stdout):
         super().__init__()
+        self.old_stdout = old_stdout
 
-        self._websocket = websocket
-        self._log_type = log_type
-        self._delegate = delegate
-
-        self._data = []
-        self._futures = []
-
-        self._loop = loop
-
-    def __enter__(self):
-        self._delegate.__enter__()
-
-        return self
-
-    def __exit__(self, exc_type, exc_val, exc_tb):
-        self._delegate.__exit__(exc_type, exc_val, exc_tb)
-
-    def __await__(self):
-        self._flush_socket()
-
-        futures = self._futures.copy()
-        self._futures.clear()
-
-        yield from asyncio.gather(*futures).__await__()
-
-    def __getattr__(self, item):
-        if item in self.__dict__:
-            return self.__dict__[item]
-
-        return getattr(self._delegate, item)
+    def write(self, text):
+        self.old_stdout.write(text)
+        if text.strip():
+            logs.put(text.strip())
+        return super().write(text)
 
     def flush(self):
-        self._delegate.flush()
-        self._flush_socket()
-
-    def _flush_socket(self):
-        coroutine = send_data(self._websocket, self._data.copy())
-        future = asyncio.ensure_future(coroutine, loop=self._loop)
-
-        self._futures.append(future)
+        self.old_stdout.flush()
+        return super().flush()
 
-        self._data.clear()
 
-    def _send_line(self, line: str):
-        self._data.append(f"{self._log_type}:{line}")
-
-    def write(self, text: str):
-        count = self._delegate.write(text)
-
-        self._send_line(text)
-
-        return count
-
-    def writelines(self, lines: Iterable[str]):
-        self._delegate.writelines(lines)
-
-        for line in lines:
-            self._send_line(line)
+sys.stdout = LogsIO(sys.stdout)
+sys.stderr = LogsIO(sys.stderr)
```

Comment:
These should still write to the `logs` queue the output prefix(`out:` and `err:`) so the validator knows where to print it